src/modules/job-manager/alloc.c

```diff
@@ -483,12 +483,14 @@ static void prep_cb (flux_reactor_t *r, flux_watcher_t *w,
 {
     struct job_manager *ctx = arg;
     struct alloc *alloc = ctx->alloc;
+    struct job *job;
 
     if (!alloc->ready || alloc->disable)
         return;
     if (alloc->mode == SCHED_SINGLE && alloc->alloc_pending_count > 0)
         return;
-    if (zlistx_first (alloc->queue))
+    if ((job = zlistx_first (alloc->queue))
+        && job->urgency != FLUX_JOB_URGENCY_HOLD)
         flux_watcher_start (alloc->idle);
```

Comment:
If a held job ends up at the top of the queue, but there are other jobs underneath, won't this fail to wake up the reactor?

Reply:
Unless I have a bug somewhere, I don't think so?  B/c the queue is sorted, if the job at the head of the queue has urgency == 0, then presumably all jobs after it are also urgency == 0.

Reply:
That makes sense, thanks, I didn't see that at first.  It might be good to add a comment to top of `prep_cb()` and `check_cb()`, something like
```c
/* The queue is sorted from highest to lowest priority, so if the
 * first job has urgency=HOLD (priority=MIN), all other jobs must have
 * the same priority, and no alloc requests can be sent.
 */
```

Reply:
For the record, I did manually verify that submitting a held job doesn't block things up, though your explanation makes it obvious that the code is correct.